src/inmanta/protocol/methods_v2.py

```diff
@@ -366,7 +366,15 @@ def get_compile_data(id: uuid.UUID) -> Optional[model.CompileData]:
 
 
 @typedmethod(
-    path="/resource_actions", operation="GET", arg_options=methods.ENV_OPTS, client_types=[ClientType.api], api_version=2
+    path="/resource_actions",
+    operation="GET",
+    arg_options={
+        "first_timestamp": ArgOption(getter=methods.datetime_utc_to_local),
```

Comment:
Why not handle (all of) this in the protocol decoder?

Reply:
e.g. in `RESTBase._decode` we could introduce a custom json deserializer cfr the custom serializer

Reply:
That looks a lot more promising than the current approach, thanks!

Reply:
I've looked into it a bit and it looks like
1. the protocol decoder is used by the Python client as well, so it would require a parameter for the decoder similar to how the encoder is implemented in this pull request.
2. I'm not sure how to cleanly perform the decoding. At that point all data is still untyped so there's no easy way to conclusively identify datetime instances.

I'd say the middle way between the approach of this pull request and converting it in the decoder would be to modify the pydantic model used by `MethodProperties.validate_arguments`, which would remove the need for `ArgOptions` to specify datetime conversion. The reason I didn't was that it felt like a bit of a hack to use pydantic validation to perform value conversion: the construction of a pydantic object would suddenly no longer be idempotent in the sense that `my_model != MyModel(**(my_model.dict()))`.

To be honest I'm having a hard time deciding which approach is best. I think I still slightly prefer the current one but only as the lesser of two evils, I still don't like it. Additionally, I'm not sure my understanding of the context is sufficiently deep to consider all consequences.
What is your opinion at this point? Would you still prefer this to be handled in the protocol decoder?